src/OrchardCore.Modules/OrchardCore.Email/Controllers/AdminController.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Email.Models;
+using OrchardCore.Email.Services;
+
+namespace OrchardCore.Email.Controllers
+{
+    public class AdminController : Controller
+    {
+        private readonly IStringLocalizer<AdminController> T;
+        private readonly IAuthorizationService _authorizationService;
+
+        public AdminController(IStringLocalizer<AdminController> stringLocalizer,
+            IAuthorizationService authorizationService)
+        {
+            T = stringLocalizer;
+            _authorizationService = authorizationService;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> SmtpSettingsTest(TestSmtpOptions testSettings)
+        {
+            if ( !await _authorizationService.AuthorizeAsync( User, Permissions.ManageEmailSettings ) )
+            {
+                return Unauthorized();
+            }
+
+            var logger = new FakeLogger<SmtpService>();
+
+            if ( !ModelState.IsValid )
+            {
+                logger.LogError( "Invalid settings" );
+                return Json( new { error = logger.Message } );
+            }
+
+            var smtp = new SmtpService( Options.Create<SmtpSettings>( testSettings ), logger );
```

Comment:
Inject SmtpService, with IOptions of SmtpSettings being injected in to that service

Reply:
Could you give me a clue on how to do that?

Reply:
In the controller's constructor, add `ISmtpService`. And in `SmtpService`'s constructor add `IOption<SmtpSettings>`. This will let the DI inject the correct instances.

Reply:
`SmtpService` already has `IOption<SmtpSettings>`, the problem here is that we need to plug in test smtp settings.  How do I have DI injected test settings?